File: Resueltos_por_competidor/Juan_Martinez/Party Monster.cpp
/*
 * Autor: Juan Martinez
 * Problema: Party Monster (2227B)
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/contest/2227/problem/B
 */

#include <bits/stdc++.h>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int n, izq, der , cant;
    cin>> n;
    char c;
    while(n--) {
        cin>>cant;
        izq = 0;
        der = 0;
        for(int i = 0; i < cant; i++) {
            cin>>c;
            if(c == '(' ) izq++;
            else der++;
        }

        if(izq == der) cout <<"YES"<< endl;
        else cout <<"NO"<< endl;
    }

    return 0;
}

File: Resueltos_por_competidor/Mario_Rodriguez/Party_Monster.java

/*
 * Autor: Mario Rodríguez
 * Problema: Party Monster
 * Juez online: CodeForces
 * Veredicto: Accepted
 * URL: https://codeforces.com/problemset/problem/2227/B
 */
import java.util.Scanner;

public class Party_Monster {

	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);
		int c = sc.nextInt();

		while (c-- > 0) {

			int c2 = sc.nextInt();
			int cin = 0, cot = 0;
			String temp = sc.next();

			for (int i = 0; i < c2; i++) {

				if (temp.charAt(i) == '(')
					cin++;
				else
					cot++;

			}
			sc.nextLine();

			System.out.println((cin == cot) ? "YES" : "NO");

		}

		sc.close();

	}

}
